### ISSUE TITLE
How to enable API by deployment manager

### ISSUE THREAD

**Issue by BALABHASKARARAO** [GoogleCloudPlatform/deploymentmanager-samples#320]
I am using this command  gcloud services enable audit.googleapis.com

would you suggest me How to enable API by deployment manager .Yaml file.

**Comment by ocsig** [comment]
TL;DR: Use the Cloud Foundation Toolkit, especially it's project template and you get functionality.

Detailed answer:
I suggest you to take a look at the project factory templates:
[Community/cloud-foundation/templates/project](https://github.com/GoogleCloudPlatform/deploymentmanager-samples/tree/master/community/cloud-foundation/templates/project)

This template and the other templates under Cloud-foundation have a lots of great practice, highly advised to take a look at them. ( More templates are in the Cloud-foundation branch and currently under review.)

To answer your question:
Enabling APIs from a DM template is possible, it's a good practice to do after the creation of the project ( which can be also done by DM.) The Yaml file only contains configuration/input parameters. The actual action is defined in the python files. Take a look into [this ](https://github.com/GoogleCloudPlatform/deploymentmanager-samples/blob/22849bd7a044f7084358397eb8c29a0d45007f60/community/cloud-foundation/templates/project/project.py#L90) section.

**Comment by BALABHASKARARAO** [comment]
Thank you ocsig. you gave information source links.
I worked on that. I get the following code to enable API by deployment mangaer.

#the below code is used to enable APi by deployment deploymentmanager
#consumerId: "project:aaa-devops"  syntax is consumerId: ""project:<project-name>"

resources:
- name: enable-computeengineapi
  action: gcp-types/servicemanagement-v1:servicemanagement.services.enable
  properties:
    consumerId: "project:<projectname>"
    serviceName: compute.googleapis.com
- name: enable-kubernetesapi
  action: gcp-types/servicemanagement-v1:servicemanagement.services.enable
  properties:
    consumerId: "project:<projectname>"
    serviceName: container.googleapis.com
